src/handlers/notebook.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import {
+  INotebookTracker,
+  NotebookPanel,
+  NotebookTracker
+} from '@jupyterlab/notebook';
+// import { CodeCell } from '@jupyterlab/cells';
+import { DebugSession } from './../session';
+import { IClientSession } from '@jupyterlab/apputils';
+import { BreakpointsService } from '../breakpointsService';
+import { CodeCell } from '@jupyterlab/cells';
+import { CellManager } from './cell';
+
+export class DebuggerNotebookTracker {
+  constructor(options: DebuggerNotebookTracker.IOptions) {
+    this.breakpointService = options.breakpointService;
+    this.notebookTracker = options.notebookTracker;
+
+    this.notebookTracker.currentChanged.connect(
+      (sender, notePanel: NotebookPanel) => {
```

Comment:
`notebookPanel` sounds more explicit. What do you think?

```suggestion
      (sender, notebookPanel: NotebookPanel) => {
```